int32     message_code
int32     agent_id
int32     targets_number
float64[] targets_list